cmd/soroban-cli/src/commands/tx/fetch/fee.rs

```diff
@@ -0,0 +1,231 @@
+use super::args;
+use crate::{
+    commands::global,
+    config::network,
+    rpc,
+    xdr::{self, Hash, SorobanTransactionMetaExt, TransactionMeta},
+};
+use clap::{command, Parser};
+use prettytable::{
+    format::{FormatBuilder, LinePosition, LineSeparator},
+    Cell, Row, Table,
+};
+use serde::{Deserialize, Serialize};
+use soroban_rpc::GetTransactionResponse;
+
+#[derive(Parser, Debug, Clone)]
+#[group(skip)]
+pub struct Cmd {
+    #[command(flatten)]
+    pub(crate) args: FeeArgs,
+}
+
+#[derive(Debug, Clone, clap::Args)]
+pub struct FeeArgs {
+    /// Transaction hash to fetch
+    #[arg(long)]
+    pub hash: Hash,
+
+    #[command(flatten)]
+    pub network: network::Args,
+
+    /// Output format for fee command
+    #[arg(long, default_value = "json")]
+    pub output: FeeOutputFormat,
+}
+
+impl FeeArgs {
```

Comment:
This could probably use some DRYing up. I'm happy to look into doing that tomorrow!